### PR TITLE
Test cases for admin packages

### PR DESCRIPTION
**Pull request by suraj-webkul**
## Issue Reference

Write test cases for admin package.

### CODE REVIEW

File: packages/Webkul/Admin/tests/AdminTestCase.php [hunk 1]
```diff
@@ -3,7 +3,20 @@
 namespace Webkul\Admin\Tests;
 
 use Tests\TestCase;
+use Webkul\User\Contracts\Admin as AdminContract;
+use Webkul\User\Models\Admin as AdminFaker;
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/AdminTestCase.php:7]: AdminModel

Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/AdminTestCase.php]: done

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    $this->loginAsAdmin();
+
+    get(route('admin.catalog.families.index'))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.families.index.title'))
+        ->assertSeeText(trans('admin::app.catalog.families.index.add'));
+});
+
+it('should return listing items of attribute family', function () {
+    // Act and Assert
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:31]: Use attribute family factory and create one more attribute family via factory. 

And remove the hard-coded assertion.

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        ->assertRedirectToRoute('admin.catalog.families.index')
+        ->isRedirection();
+
+    $this->assertDatabaseHas('attribute_families', [
+        'code' => $updatedCode,
+        'name' => $attributeFamily->name,
+    ]);
+});
+
+it('should update the existing attribute family', function () {
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:110]: delete

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    $this->loginAsAdmin();
+
+    deleteJson(route('admin.catalog.families.delete', $attributeFamily->id))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.catalog.families.delete-success'));
+
+    $this->assertDatabaseMissing('attribute_families', [
+        'id' => $attributeFamily->id,
+    ]);
+});
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:124]: Add one more testcase here,

should not be able to delete the attribute family if the attribute family is the only one present

File: packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    // Cleaning up the row  which are creating
+    CustomerModel::query()->delete();
+    OrderAddressModel::query()->delete();
+    OrderModel::query()->delete();
+    OrderPaymentModel::query()->delete();
+    CartItemModel::query()->delete();
+    CartModel::query()->delete();
+    ProductModel::query()->delete();
+    InvoiceModel::query()->delete();
+});
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php:29]: Add new line.

File: packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        'invoice_id' => $invoice->id,
+        'order_id'   => $order->id,
+    ]);
+});
+
+it('should view the transaction', function () {
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    // Arrange
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php:107]: All arrangements should be done first. It should follow AAA concept.

Arrange, Act & Assert

File: packages/Webkul/Admin/tests/Feature/Settings/LocaleTest.php [hunk 7]
```diff
@@ -0,0 +1,90 @@
+<?php
+
+use Webkul\Core\Models\Locale as LocaleModel;
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Settings/LocaleTest.php:3]: Remove all unnecessary aliases.

File: packages/Webkul/Admin/tests/Feature/Settings/TaxRatesTest.php [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    // Act and Assert
+    $this->loginAsAdmin();
+
+    get(route('admin.settings.taxes.rates.create'))
+        ->assertOk()
+        ->assertSeeText(trans('admin::app.settings.taxes.rates.create.title'))
+        ->assertSeeText(trans('admin::app.settings.taxes.rates.create.back-btn'));
+});
+
+it('should store the newly created taxe rates', function () {
```
Comment (devansh-webkul) [inline packages/Webkul/Admin/tests/Feature/Settings/TaxRatesTest.php:37]: typo

File: packages/Webkul/CMS/src/Helpers/CmsPageFaker.php [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Webkul\CMS\Helpers;
+
+use Webkul\CMS\Models\Page;
+
+class CmsPageFaker
```
Comment (devansh-webkul) [inline packages/Webkul/CMS/src/Helpers/CmsPageFaker.php:7]: No need for this class.

File: packages/Webkul/Sales/src/Database/Factories/InvoiceFactory.php [hunk 10]
```diff
@@ -60,8 +51,7 @@ public function definition(): array
             'base_tax_amount'       => $taxAmount,
             'discount_amount'       => 0,
             'base_discount_amount'  => 0,
-            'order_id'              => $attributes['order_id'],
-            'order_address_id'      => $attributes['order_address_id'],
+            'order_id'              => Order::factory(),
```
Comment (devansh-webkul) [inline packages/Webkul/Sales/src/Database/Factories/InvoiceFactory.php:54]: Remove this factory dependency. No factory will depend on each other.

File: packages/Webkul/Sales/src/Database/Factories/OrderFactory.php [hunk 11]
```diff
@@ -53,18 +53,18 @@ public function definition(): array
...
+            'grand_total_invoiced'      => $grantTotal,
+            'base_grand_total_invoiced' => $grantTotal,
+            'grand_total_refunded'      => $grantTotal,
+            'base_grand_total_refunded' => $grantTotal,
+            'sub_total'                 => $grantTotal,
+            'base_sub_total'            => $grantTotal,
+            'sub_total_invoiced'        => $grantTotal,
+            'base_sub_total_invoiced'   => $grantTotal,
+            'sub_total_refunded'        => $grantTotal,
+            'base_sub_total_refunded'   => $grantTotal,
```
Comment (devansh-webkul) [inline packages/Webkul/Sales/src/Database/Factories/OrderFactory.php:67]: grandTotal

Comment (suraj-webkul) [reply packages/Webkul/Sales/src/Database/Factories/OrderFactory.php:67]: done

File: packages/Webkul/Sales/src/Database/Factories/InvoiceFactory.php (hunk 10 above)
Comment (suraj-webkul) [reply packages/Webkul/Sales/src/Database/Factories/InvoiceFactory.php:54]: done

File: packages/Webkul/CMS/src/Helpers/CmsPageFaker.php (hunk 9 above)
Comment (suraj-webkul) [reply packages/Webkul/CMS/src/Helpers/CmsPageFaker.php:7]: delete done.

File: packages/Webkul/Admin/tests/Feature/Settings/TaxRatesTest.php (hunk 8 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Settings/TaxRatesTest.php:37]: done

File: packages/Webkul/Admin/tests/Feature/Settings/LocaleTest.php (hunk 7 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Settings/LocaleTest.php:3]: done

File: packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php (hunk 6 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php:107]: done

File: packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php (hunk 5 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Sales/TransactionTest.php:29]: done

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php (hunk 2 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:31]: done

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php (hunk 3 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:110]: done

File: packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php (hunk 4 above)
Comment (suraj-webkul) [reply packages/Webkul/Admin/tests/Feature/Catalog/AttributeFamilyTest.php:124]: done

File: packages/Webkul/Admin/src/Http/Controllers/Catalog/AttributeController.php [hunk 12]
```diff
@@ -64,7 +64,7 @@ public function store()
 
         $requestData = request()->all();
 
-        if (! $requestData['default_value']) {
+        if (! isset($requestData['default_value'])) {
```
[the remark this answers is not in the document]
Comment (suraj-webkul) [reply packages/Webkul/Admin/src/Http/Controllers/Catalog/AttributeController.php]: done

**Review by devansh-webkul — APPROVED** [review]

**Pull request merged**